### ISSUE TITLE
Reconnect ble device when disconnected

### ISSUE THREAD

**Issue by PrachiKarapurkar** [xabre/xamarin-bluetooth-le#447]
How to automatically reconnect ble device when disconnected ?

**Comment by smsissuechecker** [comment]
Hi @PrachiKarapurkar,

I'm the friendly issue checker.
It seems like (100.00 %) you haven't used our issue template :cry: I think it is very frustrating for the repository owners, if you ignore them.

If you think it's fine to make an exception, just ignore this message.
**But if you think it was a mistake to delete the template, please close the issue and create a new one.**

Thanks!

**Comment by Abesoddy** [comment]
Hello, i'm also interested to know a solution to reconnect automatically a ble device with this plugin.

Thanks !!

**Comment by dariofresu** [comment]
Have you guys solved the problem somehow?
@PrachiKarapurkar @Abesoddy

**Comment by PrachiKarapurkar** [comment]
> Have you guys solved the problem somehow?
> @PrachiKarapurkar @Abesoddy

No, haven't got any solution yet.

**Comment by dariofresu** [comment]
> > Have you guys solved the problem somehow?
> > @PrachiKarapurkar @Abesoddy
> 
> No, haven't got any solution yet.

Reading the forums I found a turn around that might help you depending on your application, but I wouldn't recommend it for production purposes.

I do the following:

1.  When creating the model subscribe to the following:   

            //Notify when there is no connection
            bleAdapter.DeviceDisconnected += OnDeviceDisconnected;
            //Notify when connection is lost
            bleAdapter.DeviceConnectionLost += OnDeviceConnectionLost;

2. During the first connection save your device in a variable:

               //connect the device
               await bleAdapter.ConnectToDeviceAsync(model.DispositiveConnected, new ConnectParameters(true), source.Token);
               BLE_device = model.DispositiveConnected;

3. For when you lose the connection and OnDeviceConnectionLost gets called then proceed as follows:

- Check if the device is disconnected
- If disconnected, clear the list of devices
- Start scanning
- Stop scanning as soon as you find the BLE_device saved previously
- Reconnect to the device
- Clear all the services 
- Look for new service 
- Clear characteristics
- Look for the new characteristics
- Subscribe to characteristic notification

Basically, you do it has if the first time you connect it.

Hope it helps,
Dario

**Comment by Jay-57blocks** [comment]
@dariofresu Thank you for sharing. Why is it not recommended for production environments? Are there better alternatives?

**Comment by Jay-57blocks** [comment]
@janusw @Abesoddy @PrachiKarapurkar @smsissuechecker Have you found a reliable solution for "reconnecting after disconnecting Bluetooth devices"? I need the reconnection logic and would appreciate your advice.
